packages/data/lib/repository/local_storage_repository.dart

```diff
@@ -20,4 +29,119 @@ class LocalStorageRepositoryImpl implements LocalStorageRepository {
       password,
     );
   }
-}
\ No newline at end of file
+
+  @override
+  Future<void> saveMoviesToCache(
+    List<MovieCache> remoteMovieList,
+    MovieType movieType,
+  ) async {
+    Batch batch = _db.batch();
+    if (remoteMovieList.isNotEmpty) {
+      for (MovieCache movie in remoteMovieList) {
+        batch.insert(
+          'Movies',
+          movie.toJson(
+            movie,
+            movieType,
+          ),
+        );
+      }
+      await batch.commit();
+    }
+  }
+
+  @override
+  Future<void> deleteByIds(
+    List<int> ids,
+    MovieType movieType,
+  ) async {
+    Batch batch = _db.batch();
+    if (ids.isNotEmpty) {
+      for (int id in ids) {
+        batch.delete(
+          'Movies',
+          where: '"tmdb" = ? and "movieType" = ?',
+          whereArgs: [id, movieType.name],
+        );
+      }
+      await batch.commit();
+    }
+  }
+
+  @override
+  Future<List<MovieCache>> getMoviesFromCache(MovieType movieType) async {
+    final List<Map<String, dynamic>> maps =
+        await _db.query('Movies WHERE movieType = "${movieType.name}"');
```

Comment:
strange query. maybe we need select, no?

Reply:
_db.query = SELECT * FROM by default

Reply:
full form seems more readable :)

Reply:
got it. in this case keep as u have